From 363996287a7eb29c0147bcc91d73a7d7ea1b52ea Mon Sep 17 00:00:00 2001
From: Liron Himi <lironh@marvell.com>
Date: Mon, 17 Dec 2018 18:04:11 +0200
Subject: [PATCH 13/30] patches: fine tuning params for l3fwd for a7k/a8k

Change l3fwd settings according to number of used cores:

l3fwd-a7_8k-c1 for 1 core:

#define MAX_PKT_BURST 256
#define RTE_TEST_RX_DESC_DEFAULT 1024
#define RTE_TEST_TX_DESC_DEFAULT 2048

l3fwd-a7_8k-c2 for 2 cores:

#define MAX_PKT_BURST 128
#define RTE_TEST_RX_DESC_DEFAULT 1024
#define RTE_TEST_TX_DESC_DEFAULT 2048

l3fwd-a7_8k-c3 for 3 cores:

#define MAX_PKT_BURST 64
#define RTE_TEST_RX_DESC_DEFAULT 1024
#define RTE_TEST_TX_DESC_DEFAULT 2048

l3fwd-a7_8k-c4 for 4 cores:

#define MAX_PKT_BURST 32
#define RTE_TEST_RX_DESC_DEFAULT 1024
#define RTE_TEST_TX_DESC_DEFAULT 2048

Change-Id: Idcc0516c8467152277734305401ebde32c1bf321
Signed-off-by: Liron Himi <lironh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61890
Reviewed-by: Alan Winkowski <walan@marvell.com>
(cherry picked from commit 37af0224717fc9a05d2e79a91f211ce55cf31fe5)
Reviewed-on: http://vgitil04.il.marvell.com:8080/61962
Reviewed-on: https://sj1git1.cavium.com/1191
Reviewed-by: Alan Winkowski
Reviewed-by: Liron Himi
Tested-by: Liron Himi
---
 ...-l3fwd-tune-params-for-1-core-for-a7k-a8k.patch | 57 ++++++++++++++++++++++
 ...l3fwd-tune-params-for-2-cores-for-a7k-a8k.patch | 57 ++++++++++++++++++++++
 ...l3fwd-tune-params-for-3-cores-for-a7k-a8k.patch | 57 ++++++++++++++++++++++
 ...l3fwd-tune-params-for-4-cores-for-a7k-a8k.patch | 43 ++++++++++++++++
 4 files changed, 214 insertions(+)
 create mode 100644 patches/0001-examples-l3fwd-tune-params-for-1-core-for-a7k-a8k.patch
 create mode 100644 patches/0001-examples-l3fwd-tune-params-for-2-cores-for-a7k-a8k.patch
 create mode 100644 patches/0001-examples-l3fwd-tune-params-for-3-cores-for-a7k-a8k.patch
 create mode 100644 patches/0001-examples-l3fwd-tune-params-for-4-cores-for-a7k-a8k.patch

diff --git a/patches/0001-examples-l3fwd-tune-params-for-1-core-for-a7k-a8k.patch b/patches/0001-examples-l3fwd-tune-params-for-1-core-for-a7k-a8k.patch
new file mode 100644
index 0000000..a2a5d00
--- /dev/null
+++ b/patches/0001-examples-l3fwd-tune-params-for-1-core-for-a7k-a8k.patch
@@ -0,0 +1,57 @@
+From 8e514d01aaeac9c5b9755e5519269f71e509fa8a Mon Sep 17 00:00:00 2001
+From: Natalie Samsonov <nsamsono@marvell.com>
+Date: Tue, 14 Aug 2018 20:02:11 +0300
+Subject: [PATCH 1/1] examples/l3fwd: tune params for 1 core for a7k/a8k
+
+Creates executable with name: l3fwd-a7_8k-c1
+
+Change-Id: Ib9b19d6427b3081bb19890f5dac5ba25eee2d9f9
+Signed-off-by: Natalie Samsonov <nsamsono@marvell.com>
+---
+ examples/l3fwd/Makefile | 2 +-
+ examples/l3fwd/l3fwd.h  | 2 +-
+ examples/l3fwd/main.c   | 2 +-
+ 3 files changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/examples/l3fwd/Makefile b/examples/l3fwd/Makefile
+index cccdd9d..915bf57 100644
+--- a/examples/l3fwd/Makefile
++++ b/examples/l3fwd/Makefile
+@@ -2,7 +2,7 @@
+ # Copyright(c) 2010-2016 Intel Corporation
+ 
+ # binary name
+-APP = l3fwd
++APP = l3fwd-a7_8k-c1
+ 
+ # all source are stored in SRCS-y
+ SRCS-y := main.c l3fwd_lpm.c l3fwd_em.c
+diff --git a/examples/l3fwd/l3fwd.h b/examples/l3fwd/l3fwd.h
+index c962dea..777dd46 100644
+--- a/examples/l3fwd/l3fwd.h
++++ b/examples/l3fwd/l3fwd.h
+@@ -15,7 +15,7 @@
+ #define NO_HASH_MULTI_LOOKUP 1
+ #endif
+ 
+-#define MAX_PKT_BURST     32
++#define MAX_PKT_BURST     256
+ #define BURST_TX_DRAIN_US 100 /* TX drain every ~100us */
+ 
+ #define MAX_RX_QUEUE_PER_LCORE 16
+diff --git a/examples/l3fwd/main.c b/examples/l3fwd/main.c
+index e7111fa..6fa5d6a 100644
+--- a/examples/l3fwd/main.c
++++ b/examples/l3fwd/main.c
+@@ -51,7 +51,7 @@
+  * Configurable number of RX/TX ring descriptors
+  */
+ #define RTE_TEST_RX_DESC_DEFAULT 1024
+-#define RTE_TEST_TX_DESC_DEFAULT 1024
++#define RTE_TEST_TX_DESC_DEFAULT 2048
+ 
+ #define MAX_TX_QUEUE_PER_PORT RTE_MAX_ETHPORTS
+ #define MAX_RX_QUEUE_PER_PORT 128
+-- 
+2.7.4
+
diff --git a/patches/0001-examples-l3fwd-tune-params-for-2-cores-for-a7k-a8k.patch b/patches/0001-examples-l3fwd-tune-params-for-2-cores-for-a7k-a8k.patch
new file mode 100644
index 0000000..da75170
--- /dev/null
+++ b/patches/0001-examples-l3fwd-tune-params-for-2-cores-for-a7k-a8k.patch
@@ -0,0 +1,57 @@
+From 6801a025cc7e9332cc7ed9ae5b2bd07d093fa3fc Mon Sep 17 00:00:00 2001
+From: Natalie Samsonov <nsamsono@marvell.com>
+Date: Tue, 14 Aug 2018 21:04:59 +0300
+Subject: [PATCH 1/1] examples/l3fwd: tune params for 2 cores for a7k/a8k
+
+Creates executable with name: l3fwd-a7_8k-c2
+
+Change-Id: Idc8285ea7e19145b78a886fcf26d6a9d038bdee9
+Signed-off-by: Natalie Samsonov <nsamsono@marvell.com>
+---
+ examples/l3fwd/Makefile | 2 +-
+ examples/l3fwd/l3fwd.h  | 2 +-
+ examples/l3fwd/main.c   | 2 +-
+ 3 files changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/examples/l3fwd/Makefile b/examples/l3fwd/Makefile
+index cccdd9d..ac137e5 100644
+--- a/examples/l3fwd/Makefile
++++ b/examples/l3fwd/Makefile
+@@ -2,7 +2,7 @@
+ # Copyright(c) 2010-2016 Intel Corporation
+ 
+ # binary name
+-APP = l3fwd
++APP = l3fwd-a7_8k-c2
+ 
+ # all source are stored in SRCS-y
+ SRCS-y := main.c l3fwd_lpm.c l3fwd_em.c
+diff --git a/examples/l3fwd/l3fwd.h b/examples/l3fwd/l3fwd.h
+index c962dea..97531c1 100644
+--- a/examples/l3fwd/l3fwd.h
++++ b/examples/l3fwd/l3fwd.h
+@@ -15,7 +15,7 @@
+ #define NO_HASH_MULTI_LOOKUP 1
+ #endif
+ 
+-#define MAX_PKT_BURST     32
++#define MAX_PKT_BURST     128
+ #define BURST_TX_DRAIN_US 100 /* TX drain every ~100us */
+ 
+ #define MAX_RX_QUEUE_PER_LCORE 16
+diff --git a/examples/l3fwd/main.c b/examples/l3fwd/main.c
+index e7111fa..6fa5d6a 100644
+--- a/examples/l3fwd/main.c
++++ b/examples/l3fwd/main.c
+@@ -51,7 +51,7 @@
+  * Configurable number of RX/TX ring descriptors
+  */
+ #define RTE_TEST_RX_DESC_DEFAULT 1024
+-#define RTE_TEST_TX_DESC_DEFAULT 1024
++#define RTE_TEST_TX_DESC_DEFAULT 2048
+ 
+ #define MAX_TX_QUEUE_PER_PORT RTE_MAX_ETHPORTS
+ #define MAX_RX_QUEUE_PER_PORT 128
+-- 
+2.7.4
+
diff --git a/patches/0001-examples-l3fwd-tune-params-for-3-cores-for-a7k-a8k.patch b/patches/0001-examples-l3fwd-tune-params-for-3-cores-for-a7k-a8k.patch
new file mode 100644
index 0000000..6c11b42
--- /dev/null
+++ b/patches/0001-examples-l3fwd-tune-params-for-3-cores-for-a7k-a8k.patch
@@ -0,0 +1,57 @@
+From 16401bf96199ef07fc183368259535190765f0db Mon Sep 17 00:00:00 2001
+From: Natalie Samsonov <nsamsono@marvell.com>
+Date: Tue, 14 Aug 2018 21:32:55 +0300
+Subject: [PATCH 1/1] examples/l3fwd: tune params for 3 cores for a7k/a8k
+
+Creates executable with name: l3fwd-a7_8k-c3
+
+Change-Id: I479303d8b09fa7fb6330c4bca47d20347e4ff382
+Signed-off-by: Natalie Samsonov <nsamsono@marvell.com>
+---
+ examples/l3fwd/Makefile | 2 +-
+ examples/l3fwd/l3fwd.h  | 2 +-
+ examples/l3fwd/main.c   | 2 +-
+ 3 files changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/examples/l3fwd/Makefile b/examples/l3fwd/Makefile
+index cccdd9d..1f11d3a 100644
+--- a/examples/l3fwd/Makefile
++++ b/examples/l3fwd/Makefile
+@@ -2,7 +2,7 @@
+ # Copyright(c) 2010-2016 Intel Corporation
+ 
+ # binary name
+-APP = l3fwd
++APP = l3fwd-a7_8k-c3
+ 
+ # all source are stored in SRCS-y
+ SRCS-y := main.c l3fwd_lpm.c l3fwd_em.c
+diff --git a/examples/l3fwd/l3fwd.h b/examples/l3fwd/l3fwd.h
+index c962dea..bbe1e7f 100644
+--- a/examples/l3fwd/l3fwd.h
++++ b/examples/l3fwd/l3fwd.h
+@@ -15,7 +15,7 @@
+ #define NO_HASH_MULTI_LOOKUP 1
+ #endif
+ 
+-#define MAX_PKT_BURST     32
++#define MAX_PKT_BURST     64
+ #define BURST_TX_DRAIN_US 100 /* TX drain every ~100us */
+ 
+ #define MAX_RX_QUEUE_PER_LCORE 16
+diff --git a/examples/l3fwd/main.c b/examples/l3fwd/main.c
+index e7111fa..6fa5d6a 100644
+--- a/examples/l3fwd/main.c
++++ b/examples/l3fwd/main.c
+@@ -51,7 +51,7 @@
+  * Configurable number of RX/TX ring descriptors
+  */
+ #define RTE_TEST_RX_DESC_DEFAULT 1024
+-#define RTE_TEST_TX_DESC_DEFAULT 1024
++#define RTE_TEST_TX_DESC_DEFAULT 2048
+ 
+ #define MAX_TX_QUEUE_PER_PORT RTE_MAX_ETHPORTS
+ #define MAX_RX_QUEUE_PER_PORT 128
+-- 
+2.7.4
+
diff --git a/patches/0001-examples-l3fwd-tune-params-for-4-cores-for-a7k-a8k.patch b/patches/0001-examples-l3fwd-tune-params-for-4-cores-for-a7k-a8k.patch
new file mode 100644
index 0000000..184f56f
--- /dev/null
+++ b/patches/0001-examples-l3fwd-tune-params-for-4-cores-for-a7k-a8k.patch
@@ -0,0 +1,43 @@
+From 8c56c58849eb8c6ae54321274ec2e7f2947c11ee Mon Sep 17 00:00:00 2001
+From: Natalie Samsonov <nsamsono@marvell.com>
+Date: Tue, 14 Aug 2018 21:50:40 +0300
+Subject: [PATCH 1/1] examples/l3fwd: tune params for 4 cores for a7k/a8k
+
+Creates executable with name: l3fwd-a7_8k-c4
+
+Change-Id: I1a33be8e10b12c00cccba326c797b4066cb18cc2
+Signed-off-by: Natalie Samsonov <nsamsono@marvell.com>
+---
+ examples/l3fwd/Makefile | 2 +-
+ examples/l3fwd/main.c   | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/examples/l3fwd/Makefile b/examples/l3fwd/Makefile
+index cccdd9d..2d50675 100644
+--- a/examples/l3fwd/Makefile
++++ b/examples/l3fwd/Makefile
+@@ -2,7 +2,7 @@
+ # Copyright(c) 2010-2016 Intel Corporation
+ 
+ # binary name
+-APP = l3fwd
++APP = l3fwd-a7_8k-c4
+ 
+ # all source are stored in SRCS-y
+ SRCS-y := main.c l3fwd_lpm.c l3fwd_em.c
+diff --git a/examples/l3fwd/main.c b/examples/l3fwd/main.c
+index e7111fa..6fa5d6a 100644
+--- a/examples/l3fwd/main.c
++++ b/examples/l3fwd/main.c
+@@ -51,7 +51,7 @@
+  * Configurable number of RX/TX ring descriptors
+  */
+ #define RTE_TEST_RX_DESC_DEFAULT 1024
+-#define RTE_TEST_TX_DESC_DEFAULT 1024
++#define RTE_TEST_TX_DESC_DEFAULT 2048
+ 
+ #define MAX_TX_QUEUE_PER_PORT RTE_MAX_ETHPORTS
+ #define MAX_RX_QUEUE_PER_PORT 128
+-- 
+2.7.4
+
-- 
2.7.4

